src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -1013,9 +1009,10 @@ static ucs_status_t uct_tcp_sockcm_ep_server_create(uct_tcp_sockcm_ep_t *tcp_ep,
 
     status = uct_tcp_sockcm_ep_pack_priv_data(tcp_ep, priv_data,
                                               priv_data_length);
+    if (status == UCS_OK) {
+        tcp_ep->state |= UCT_TCP_SOCKCM_EP_SERVER_CREATED;
```

Comment:
why moved to here? it's under lock anyway

Reply:
because before the PR , in case of err, dtor was called outside of the lock, it caused wrong err callback, I moved setting the flag to here to make the code clearer at least since we have to invoke err callback *only* on successfully created EP owned by user

Reply:
>dtor was called outside of the lock, it caused wrong err callback

what was wrong? which callback was called?

>we have to invoke err callback only on successfully created EP owned by user

how does the location of setting this flag affect this?
we can get to the err label before this flag is set

Reply:
discussed offline - can't call the err cb before this flag is set. too early setting of it may result in just that in case of an error - when the client exits and the server's async thread calls the err_cb.